vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -1367,11 +1367,13 @@ protected ArrowBuf allocateOrGetLastDataBuffer(int length) {
   protected final void setBytes(int index, byte[] value, int start, int length) {
     int writePosition = index * ELEMENT_SIZE;
 
-    // to clear the memory segment of view being written to
-    // this is helpful in case of overwriting the value
-    viewBuffer.setZero(writePosition, ELEMENT_SIZE);
-
     if (length <= INLINE_SIZE) {
+      // to clear the memory segment of view being written to
+      // if it has been set
+      if (viewBuffer.getLong(writePosition) != 0 || viewBuffer.getLong(writePosition + 8) != 0) {
```

Comment:
So the basic effect is that if the view buffer is all 0, we skip setting it back to 0? And basically, for freshly allocated memory, this skips a redundant memset? Can we explain it a little more in the comment?

Reply:
Yes, it's redundant to setZero if the data is already zero. Checking will introduce some overhead, but it is much smaller than that of setMemory.

Reply:
Update in 16c9e2f4bc5921a4d6e0572954b19b5f958dbf85

Reply:
Is it safe to check only these positions? IIUC, ZERO is a valid value to be set.

Reply:
@wgtmac are we on the same page here?

Reply:
Yes, it sounds good.